src/components/Attachment/components/ProgressBar.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+
+export type ProgressBarProps = {
+  /** Progress expressed in fractional number value btw 0 and 100. */
+  progress: number;
+} & Pick<React.ComponentProps<'div'>, 'onClick'>;
+
+export const ProgressBar = ({ onClick, progress }: ProgressBarProps) => (
+  <div
+    className='str-chat__message-attachment-audio-widget--progress-track'
+    data-progress={progress}
+    data-testid='audio-progress'
+    onClick={onClick}
+    role='progressbar'
+    style={{
+      background: `linear-gradient(
+		 to right,
+		 var(--str-chat__primary-color),
+		 var(--str-chat__primary-color) ${progress}%,
+		 var(--str-chat__disabled-color) ${progress}%,
+		 var(--str-chat__disabled-color)
+	  )`,
+    }}
```

Comment:
Since we are trying to use less inline styles, can we just pass `progress` to a CSS variable, and move the rest to `stream-chat-css`?

```
style={{ '--progress': progress }}
```

Reply:
Adjusted in https://github.com/GetStream/stream-chat-react/pull/2311/commits/92fd08a14f0e4e8a680441c7d86b61cfd3334fa6